lib/IgnorePlugin.js

```diff
@@ -27,10 +39,13 @@ class IgnorePlugin {
 	 * and the resource given matches the regexp.
 	 */
 	checkResource(resource) {
-		if (!this.resourceRegExp) {
+		if (this.options.checkResource) {
```

Comment:
This doesn't handle the case where `checkResource` and `resourceRegexp` is provided.

Either forbid this case via schema or handle it.

Reply:
Same for context